### PR TITLE
Rename errc::OVERFLOW to avoid macro collisions

### PR DESCRIPTION
**Pull request by a-hirota**
## Description

`cudf::errc::OVERFLOW` conflicts with existing OVERFLOW macros -- glibc's math.h still defines one as a legacy SVID constant -- so any translation unit with such a macro in scope had the enumerator rewritten by the preprocessor and cudf/errc.hpp failed to parse. A downstream engine consuming installed libcudf headers hit exactly this.

Rename the enumerator to `ARITHMETIC_OVERFLOW` and update all internal uses (the only ones are in detail/operators/checked_arithmetic.cuh). Unlike a header-local macro guard, the rename also lets callers name the enumerator while the unrelated macro remains defined.

Add a compile regression test that defines OVERFLOW before including the header and checks both that the header parses and that the includer's macro is left untouched.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by rjzamora** [comment]
/ok to test [e8eb185](https://github.com/rapidsai/cudf/pull/23509/commits/e8eb1857e8c229583e27202bba3f6cb0b2b536a9)

File: cpp/tests/utilities_tests/errc_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+static_assert(OVERFLOW == 3, "the includer's macro must be left untouched");
+static_assert(static_cast<int>(cudf::errc::ARITHMETIC_OVERFLOW) == 1);
+
+TEST(ErrcTest, EnumeratorsSurviveMacroCollision)
+{
+  EXPECT_STREQ(cudf::to_string(cudf::errc::ARITHMETIC_OVERFLOW), "ARITHMETIC_OVERFLOW");
+  EXPECT_STREQ(cudf::to_string(cudf::errc::SUCCESS), "SUCCESS");
+  EXPECT_STREQ(cudf::to_string(cudf::errc::DIVISION_BY_ZERO), "DIVISION_BY_ZERO");
+}
```
Comment (mhaseeb123) [inline cpp/tests/utilities_tests/errc_tests.cpp]: I think this test maybe redundant and can be removed.

File: cpp/tests/CMakeLists.txt [hunk 2]
```diff
@@ -447,6 +447,7 @@ ConfigureTest(
   utilities_tests/column_wrapper_tests.cpp
   utilities_tests/context_tests.cpp
   utilities_tests/default_stream_tests.cpp
+  utilities_tests/errc_tests.cpp
```
Comment (mhaseeb123) [inline cpp/tests/CMakeLists.txt]: Same

**Review by mhaseeb123 — APPROVED** [review]
Simple rename changes look good to me. Requesting a review from @lamarrr to confirm

**Review by mhaseeb123 — CHANGES_REQUESTED** [review]
Whoops, requesting changes related to removal of unnecessary tests

**Review by lamarrr — APPROVED** [review]

File: cpp/include/cudf/errc.hpp [hunk 3]
```diff
@@ -24,9 +24,9 @@ namespace CUDF_EXPORT cudf {
  * @brief An enumeration of error codes that can occur during operations.
  */
 enum class [[nodiscard]] errc : cuda::std::int8_t {
-  SUCCESS          = 0,
-  OVERFLOW         = 1,
-  DIVISION_BY_ZERO = 2,
+  SUCCESS             = 0,
+  ARITHMETIC_OVERFLOW = 1,
```
Comment (PointKernel) [inline cpp/include/cudf/errc.hpp:28]: The new name is a bit wordy, but it doesn't seem like we have a better alternative anyway.

**Review by PointKernel — APPROVED** [review]
A question for the team: would using lowercase enumerators in cases like this make life much easier?

**Review by mhaseeb123 — APPROVED** [review]
